### PR TITLE
coding standards for slack repo 

### PR DESCRIPTION
**Pull request by gaurav1068**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "require-dev": {
         "phpunit/phpunit": "4.2.*",
         "mockery/mockery": "0.9.*",
-        "illuminate/queue": "4.*"
```
Comment (venkatvghub) [inline composer.json]: I do not see this requirement on the API side. Do we need this here?

File: src/Client.php [hunk 2]
```diff
@@ -1,558 +1,614 @@
...
+    /**
+     * Pass any unhandled methods through to a new Message
+     * instance
+     *
+     * @param string $name      The name of the method
+     * @param array  $arguments The method arguments
+     *
+     * @return \Maknz\Slack\Message
+    */
+    public function __call($name, $arguments)
```
Comment (venkatvghub) [inline src/Client.php]: Now that we are refactoring this almost. Can we typehint things pliss?

Comment (venkatvghub) [inline src/Client.php]: Additionally, update the phpdoc accordingly.

**Comment by venkatvghub** [comment]
Can we additionally ensure that we have both the sync and queue modes tested thoroughly. Also, can you add tests for these along with testing on the retry count to ensure it is respected?

File: src/Client.php (hunk 2 above)
Comment (gaurav1068) [inline src/Client.php]: I agree, this should be done but can this be done in separate PR

File: composer.json (hunk 1 above)
Comment (gaurav1068) [inline composer.json]: yes this is required, API dont have direct dependency but it is installed indirectly via dependency of laravel/framework(I am not sure exactly(

**Comment by gaurav1068** [comment]
opened two issues 
https://github.com/razorpay/slack/issues/6
https://github.com/razorpay/slack/issues/5
merging this PR. Tested on beta

**Comment by gaurav1068** [comment]
@venkatvghub tested on beta environment can you please approve this PR
